lib/reek/smells/unneeded_suppression.rb

```diff
@@ -0,0 +1,39 @@
+# frozen_string_literal: true
+
+module Reek
+  module Smells
+    #
+    # An unneeded suppression is a Reek's comment disabling a unexistent smell.
+    #
+    # See {file:docs/Unneeded-Suppression.md} for details.
+    class UnneededSuppression
+      def initialize(context, smell_detector)
+        @context = context
+        @smell_detector = smell_detector
+      end
+
+      def sniff
```

Comment:
I would move the check for the condition 'disabled but not smelly' back up into `SmellDetector#sniff`, end create a smell warning right there, passing a light-weight smell detector like object to provide the smell warning with what it needs.